File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/strategy/calculator/CalculationStrategy.java
package com.bigboxcode.designpattern.strategy.calculator;

public class CalculationStrategy {
    private final Calculation calculation;

    public CalculationStrategy(Calculation calculation) {
        this.calculation = calculation;
    }

    public float execute(float num1, float num2) {
        return this.calculation.execute(num1, num2);
    }
}

File: design-patterns/php/strategy/mathcalculation/CalculationStrategy.php
<?php
// CalculationStrategy.php

namespace BigBoxCode\DesignPattern\Strategy\MathCalculation;


class CalculationStrategy {
    public function __construct(private Calculation $calculation) {
    }

    public function execute(float $num1, float $num2): float {
        return $this->calculation->execute($num1, $num2);
    }
}